cmd/device-definitions-api/populate_device_features.go

```diff
@@ -0,0 +1,123 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/DIMO-Network/device-definitions-api/internal/config"
+	"github.com/DIMO-Network/device-definitions-api/internal/infrastructure/db/models"
+	elastic "github.com/DIMO-Network/device-definitions-api/internal/infrastructure/elasticsearch"
+	elasticModels "github.com/DIMO-Network/device-definitions-api/internal/infrastructure/elasticsearch/models"
+	"github.com/DIMO-Network/shared/db"
+	"github.com/rs/zerolog"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+)
+
+type jsonObj map[string]interface{}
+
+func prepareFeatureData(f map[string]map[string]int) []elasticModels.DeviceIntegrationFeatures {
+	ft := []elasticModels.DeviceIntegrationFeatures{}
+
+	for k, v := range f {
+		var supportLevel int8
+
+		if v["doc_count"] > 0 {
+			supportLevel = 2
+		}
+
+		feat := elasticModels.DeviceIntegrationFeatures{
+			FeatureKey:   k,
+			SupportLevel: int8(supportLevel),
+		}
+
+		ft = append(ft, feat)
+	}
+
+	return ft
+}
+
+func getIntegrationFeatures(ctx context.Context, d db.Store) (string, error) {
+	ifeats, err := models.IntegrationFeatures().All(ctx, d.DBS().Reader)
+	if err != nil {
+		return "", err
+	}
+
+	filters := jsonObj{}
+
+	for _, v := range ifeats {
+		esKey := v.ElasticProperty
+		if v.FeatureKey == "tires" {
+			esKey = v.FeatureKey
+		}
+		filters[esKey] = jsonObj{"exists": jsonObj{"field": "data." + v.ElasticProperty}}
+	}
+
+	esFilters, err := json.Marshal(filters)
+	if err != nil {
+		return "", err
+	}
+
+	return string(esFilters), nil
+}
+
+func populateDeviceFeaturesFromEs(ctx context.Context, logger zerolog.Logger, s *config.Settings) error {
```

Comment:
Consider moving this method to the top as it does the meat of the operations, and the other methods act more like helpers / refactor out logic .